File: Day01/day01part1.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class day01part1{
    public static void main(String[] args) {
        String filePath = "./day01.txt"; 

        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
            String line;
            int sum = 0;
            while ((line = br.readLine()) != null) {
                int index = 0;
                String totstring = "";

                while(!Character.isDigit(line.charAt(index))){
                    index++;
                }

                totstring+=line.charAt(index);
                index = line.length()-1;

                while(!Character.isDigit(line.charAt(index))){
                    index--;
                }
                
                totstring+=line.charAt(index);

                sum += Integer.parseInt(totstring);
                System.out.println();

            }

            System.out.println("Part 1: " + sum);

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: Day01/day01part1.rs
use std::fs::File;
use std::io::{self, BufRead};
use std::path::Path;

fn main() {
    let mut sum = 0;
    if let Ok(lines) = read_lines("./day01.txt"){
        for line in lines{
            if let Ok(ip) = line{
                let mut index = 0;
                let mut tot = String::new();

                while !ip.as_bytes()[index].is_ascii_digit() {
                    index+=1;
                }

                if let Some(character) = ip.chars().nth(index) {
                    tot.push(character);
                } else {
                    println!("Index out of range or no character found at index");
                }

                index = ip.len()-1;

                while !ip.as_bytes()[index].is_ascii_digit() {
                    index-=1;
                }

                if let Some(character) = ip.chars().nth(index) {
                    tot.push(character);
                } else {
                    println!("Index out of range or no character found at index");
                }

                if let Ok(result) = tot.parse::<usize>() {
                    sum += result;
                }
            }
        }

        println!("\nPart 1: {}", sum);
    }
}

fn read_lines<P>(filename: P) -> io::Result<io::Lines<io::BufReader<File>>>
where P: AsRef<Path>, {
    let file = File::open(filename)?;
    Ok(io::BufReader::new(file).lines())
}